src/v/cluster/members_backend.cc

```diff
@@ -1180,6 +1197,79 @@ void members_backend::handle_reallocation_finished(model::node_id id) {
     });
 }
 
+ss::future<> members_backend::reconcile_raft0_updates() {
+    while (!_as.local().abort_requested()) {
+        co_await _new_updates.wait([this] { return !_raft0_updates.empty(); });
+
+        auto update = _raft0_updates.front();
+        vlog(clusterlog.trace, "processing raft 0 update: {}", update);
+        auto err = co_await update_raft0_configuration(update);
+        if (err) {
+            vlog(
+              clusterlog.trace,
+              "raft 0 update {} returned an error - {}",
+              update,
+              err.message());
+            co_await ss::sleep_abortable(200ms, _as.local());
+            continue;
+        }
+
+        _raft0_updates.pop_front();
+    }
+}
+
+ss::future<std::error_code> members_backend::update_raft0_configuration(
+  const members_manager::node_update& update) {
+    model::revision_id revision(update.offset);
+    auto cfg = _raft0->config();
+    if (cfg.revision_id() > model::revision_id(update.offset)) {
+        co_return errc::success;
+    }
```

Comment:
Should we skip this if the Raft config has brokers? Wondering what happens if we're playing an older legacy Raft config (that is already in `_raft0`) that then adds a node update to this queue in response.

Reply:
good point, i will add this optimization, now it won't do anything as the broker is already either added or removed from raft0 configuration but we may exit early from this function.